code-experiments/src/c_linear.c

```diff
@@ -217,9 +217,17 @@ static coco_problem_t *c_linear_shuffle(coco_problem_t *problem_c,
   
   /* Pick up a random constraint number other than 1.
    * formula for U[M,N]: 
-   * M + rand() / (RAND_MAX / (N - M + 1) + 1)
+   * random = M + (rand() / (RAND_MAX + 1.0)) * (N - M + 1)
```

Comment:
this formula looks still wrong to me, in particular `N - M + 1` must AFAICS read `N - M`